docs/source/concepts/interoperability.rst

```diff
@@ -0,0 +1,464 @@
+
+  #. If `X` is not empty, then `X` contains a **Create** or **Enter** action.
+     This action precedes all other actions in `X`.
+
+  #. `X` contains at most one **Create** action.
+     If so, this action precedes all other actions in `X`.
+
+  #. If `X` contains a consuming **Exercise** action `act`, then `act` follows all other actions in `X` in `G`\ 's action order.
+
+  #. For every transfer action `act` in `X`, every other action `act'` in `X` either precedes or follows `act` in `G`\ 's action order.
+
+  #. For every maximal chain in `X` (i.e., maximal totally ordered subset of `X`), the sequence of `(`\ incoming ledger, outgoing ledger\ `)` pairs is a ledger trace, using `NONE` if the action does not have an incoming or outgoing ledger annotation.
+
+The first three conditions mimick the conditions of :ref:`causal consistency <def-causal-consistency-contract>` for ordinary causality graphs.
+They ensure that **Create** actions come first and consuming **Exercise** actions last.
+An **Enter** action takes the role of a **Create** if there is no **Create**.
+The fourth condition ensures that all transfer actions are synchronization points for a contract.
+The last condition about ledger traces ensures that contracts reside on only one DAML ledger and all usages happen on the ledger of residence.
+In particular, the next contract action after a **Leave** must be an **Enter**.
+
+For example, the above :ref:`interoperable causality graph with transfer actions <interoperable-causality-graph-linear>` is interoperable consistent for `c`.
+In particular, there is only one maximal chain in the actions on `c`, namely
+
+  **Create** `c` -> `tf1` -> **ExeN** `B` `c` `ch1` -> `tf2` -> **ExeN** `B` `c` `ch2` -> `tf3` -> **ExeN** `B` `c` `ch3`,
+
+and for each edge `act`:sub:`1` -> `act`:sub:`2`, the outgoing ledger color of `act`:sub:`1` is the same as the incoming ledger color of `act`:sub:`2`.
+The restriction to maximal chains ensures that no node is skipped.
+For example, the (non-maximal) chain
+
+  **Create** `c` -> **ExeN** `B` `c` `ch1` -> `tf2` -> **ExeN** `B` `c` `ch2` -> `tf3` -> **Exe** `B` `c` `ch3`
+
+is not a ledger trace because the outgoing ledger of the **Create** action (yellow) is not the same as the incoming ledger of the non-consuming **Exercise** action for `ch1` (green).
+Accordingly, the subgraph without the `tf1` vertex is not interoperable consistent for `c` even though it is an interoperable causality graph.
+
+Definition »Consistency for an interoperable causality graph«
+  Let `X` be a subset of actions in an interoperable causality graph `G`.
+  Then `G` is **interoperable consistent** for `X` (or `X`-**interoperable consistent**)
+  if `G` is interoperable consistent for all contracts `c` on the set of actions on `c` in `X`.
+  `G` is **interoperable consistent** if `G` is interoperable consistent on all the actions in `G`.
+
+.. note::
+   There is no interoperable consistency requirement for contract keys.
```

Comment:
I think we should say here that there is no interoperable consistency for contract keys _yet_, because we haven't decided on the uniqueness guarantees in interoperable contexts yet.

It would be easy to extend this model with domain uniqueness where Enter and Leave are also key operations, for example.

Reply:
Added `yet`.